dexmaker-tests/src/androidTest/java/com/android/dx/stock/ProxyBuilderTest.java

```diff
@@ -1239,4 +1243,34 @@ public Object invoke(Object proxy, Method method, Object[] args) throws Throwabl
         assertEquals("fake E", proxy.returnE());
 
     }
+
+    @Test
+    public void testImplementingDifferentInterfacesWithSharedClassLoader() throws IOException {
+        assumeTrue(Build.VERSION.SDK_INT >= 24);
+
+        Class<?> c1 = ProxyBuilder.forClass(SimpleClass.class)
+                .implementing(Runnable.class).withSharedClassLoader().buildProxyClass();
+        Class<?> c2 = ProxyBuilder.forClass(SimpleClass.class)
+                .implementing(Callable.class).withSharedClassLoader().buildProxyClass();
+        assertNotSame(c1, c2);
+    }
+
+    @Test
+    public void testImplementingDifferentInterfacesWithDexMakerCaching() throws Exception {
+        proxyFor(SimpleClass.class).implementing(Runnable.class).buildProxyClass();
+        assertEquals(1, getGeneratedProxyClasses().size());
```

Comment:
Why do we care how the cache is internally organized?

Reply:
I am not against this, but seems an odd thing to check in this test.
We really care that the two classes are different, not which files are generated.

Reply:
Deleted a test to check the cache